### PR TITLE
Fixing receipt logs

### PR DESCRIPTION
**Pull request by bzawisto**
## Description
This PR fixes some issues with logs attached to receipts. Evm-ds returns data as array of bytes whereas it should be exposed as hex string instead. Adding also two tests checking for generated events. 

## Backward Compatibility

- [X] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [X] **ready for review**

### Integration Test (Core Team)

- [X] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: tests/EvmAcceptanceTests/test/ContractInteraction.js [hunk 1]
```diff
@@ -149,6 +149,17 @@ describe("Contract Interaction", function () {
         expect(await contract.methods.setAddress(ADDRESS).send(options)).to.be.not.null;
         expect(await contract.methods.someAddress().call()).to.be.eq(ADDRESS);
       });
+
+      it("Calling a method on a contract that generates event without parameters", async function () {
```
Comment (its-saeed) [inline tests/EvmAcceptanceTests/test/ContractInteraction.js:153]: please change `it` description based on our style. we already have `Should have the event argument in returned object when arg type is uint256` `Should have the event name in returned object` and ...

**Comment by valeryz** [comment]
I think we should better make evm-ds return a hex string or something. it is rather easy.

File: tests/EvmAcceptanceTests/test/ContractInteraction.js (hunk 1 above)
Comment (bzawisto) [reply tests/EvmAcceptanceTests/test/ContractInteraction.js]: @its-saeed Done.

**Comment by bzawisto** [comment]
> 

Moved the fix from cpp to rust code.

**Review by its-saeed — APPROVED** [review]

**Review by CSideSteve — APPROVED** [review]

**Pull request merged**